src/eth/primitives/log_mined.rs

```diff
@@ -58,6 +58,22 @@ impl LogMined {
     }
 }
 
+// -----------------------------------------------------------------------------
+// Conversions: Other -> Self
+// -----------------------------------------------------------------------------
+impl From<EthersLog> for LogMined {
+    fn from(value: EthersLog) -> Self {
+        Self {
```

Comment:
Should we panic here? Why not impl TryFrom?

Reply:
Both approaches seems wrong. We need to ensure at the type level the input of this function have all fields populated, because they should have. They are logs that were added to a block, so these fields should never be None.